// create-plasmic-app/dist/nextjs/templates/patches/@plasmicapp+cli+0.1.330.patch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.makePlasmicPatch_plasmicapp_cli = void 0;
function makePlasmicPatch_plasmicapp_cli() {
    return `diff --git a/node_modules/@plasmicapp/cli/dist/actions/sync.d.ts b/node_modules/@plasmicapp/cli/dist/actions/sync.d.ts
index 9ac892e..f696f62 100644
--- a/node_modules/@plasmicapp/cli/dist/actions/sync.d.ts
+++ b/node_modules/@plasmicapp/cli/dist/actions/sync.d.ts
@@ -2,6 +2,7 @@ import { CommonArgs } from "..";
 import { Metadata } from "../utils/get-context";
 export interface SyncArgs extends CommonArgs {
     projects: readonly string[];
+    host: string;
     forceOverwrite: boolean;
     yes?: boolean;
     force?: boolean;
diff --git a/node_modules/@plasmicapp/cli/dist/index.js b/node_modules/@plasmicapp/cli/dist/index.js
index c2e64c2..927b765 100755
--- a/node_modules/@plasmicapp/cli/dist/index.js
+++ b/node_modules/@plasmicapp/cli/dist/index.js
@@ -502269,11 +502269,14 @@ var PlasmicApi = class {
     return this.auth.host;
   }
   get codegenHost() {
+    let host = "";
     if (!this.auth.host || this.auth.host === DEFAULT_HOST) {
-      return "https://codegen.plasmic.app";
+      host = "https://codegen.plasmic.app";
     } else {
-      return this.auth.host;
+      host = this.auth.host;
     }
+    console.log('__DEV__:codegenHost', { auth: this.auth, host, DEFAULT_HOST });
+    return host;
   }
 };
 
@@ -502450,7 +502453,9 @@ function failAuth() {
   );
 }
 async function getOrStartAuth(opts) {
+  console.log('__DEV__:getOrStartAuth', { opts });
   let auth2 = await getCurrentAuth(opts.auth);
+  console.log('__DEV__:getOrStartAuth', { auth2 });
   if (!auth2 && opts.enableSkipAuth) {
     return;
   }
@@ -503432,7 +503437,7 @@ async function getContext(args, {
 } = {}) {
   if (!args.baseDir)
     args.baseDir = process.cwd();
-  const auth2 = enableSkipAuth ? await getCurrentOrDefaultAuth(args) : await getOrInitAuth(args);
+  const auth2 = enableSkipAuth ? await getCurrentOrDefaultAuth({ ...args, enableSkipAuth }) : await getOrInitAuth(args);
   if (auth2.host.startsWith(DEFAULT_HOST)) {
     Sentry.init({
       dsn: "https://3ed4eb43d28646e381bf3c50cff24bd6@o328029.ingest.sentry.io/5285892"
@@ -503480,12 +503485,13 @@ async function getContext(args, {
   return context;
 }
 async function getCurrentOrDefaultAuth(args) {
-  const auth2 = await getCurrentAuth(args.auth);
+  // const auth2 = await getCurrentAuth(args.auth);
+  const auth2 = await getOrStartAuth(args);
   if (auth2) {
     return auth2;
   }
   return {
-    host: DEFAULT_HOST,
+    host: args.host || DEFAULT_HOST,
     user: "",
     token: ""
   };
@@ -505802,6 +505808,10 @@ function configureSyncArgs(yags, includeQuietOption = true) {
     describe: "One or more projects to sync, separated by comma. Version constraints can be specified using @. Example: projectid, projectid@>=version",
     type: "array",
     default: []
+  }).option("host", {
+    describe: "Plasmic host to use",
+    type: "string",
+    default: "https://studio.plasmic.app"
   }).option("force", {
     type: "boolean",
     describe: "Force sync to bypass specified version ranges.",
diff --git a/node_modules/@plasmicapp/cli/dist/utils/get-context.d.ts b/node_modules/@plasmicapp/cli/dist/utils/get-context.d.ts
index e9534eb..d5b9980 100644
--- a/node_modules/@plasmicapp/cli/dist/utils/get-context.d.ts
+++ b/node_modules/@plasmicapp/cli/dist/utils/get-context.d.ts
@@ -1,7 +1,7 @@
 import { CommonArgs } from "../index";
 import { PlasmicContext, PlasmicLock } from "./config-utils";
 export declare function readLock(lockFile: string): PlasmicLock;
-export declare function getContext(args: CommonArgs, { enableSkipAuth, skipMissingFiles, skipInit, }?: {
+export declare function getContext(args: CommonArgs & { host: string }, { enableSkipAuth, skipMissingFiles, skipInit, }?: {
     enableSkipAuth?: boolean;
     skipMissingFiles?: boolean;
     skipInit?: boolean;
@@ -9,7 +9,7 @@ export declare function getContext(args: CommonArgs, { enableSkipAuth, skipMissi
 /**
  * Use empty user/token to signify no auth (only returning to provide a default host).
  */
-export declare function getCurrentOrDefaultAuth(args: CommonArgs): Promise<import("./config-utils").AuthConfig>;
+export declare function getCurrentOrDefaultAuth(args: CommonArgs & { host: string; enableSkipAuth?: boolean }): Promise<import("./config-utils").AuthConfig>;
 /**
  * Table of where this metadata will be set
  *
diff --git a/node_modules/@plasmicapp/cli/src/actions/sync.ts b/node_modules/@plasmicapp/cli/src/actions/sync.ts
index 06d5710..e8bddf6 100644
--- a/node_modules/@plasmicapp/cli/src/actions/sync.ts
+++ b/node_modules/@plasmicapp/cli/src/actions/sync.ts
@@ -61,6 +61,7 @@ import { upsertStyleTokens } from "./sync-styles";
 
 export interface SyncArgs extends CommonArgs {
   projects: readonly string[];
+  host: string;
   forceOverwrite: boolean;
   yes?: boolean;
   force?: boolean;
diff --git a/node_modules/@plasmicapp/cli/src/api.ts b/node_modules/@plasmicapp/cli/src/api.ts
index 058fe45..7f1da6e 100644
--- a/node_modules/@plasmicapp/cli/src/api.ts
+++ b/node_modules/@plasmicapp/cli/src/api.ts
@@ -544,10 +544,13 @@ export class PlasmicApi {
   }
 
   private get codegenHost() {
+    let host = "";
     if (!this.auth.host || this.auth.host === DEFAULT_HOST) {
-      return "https://codegen.plasmic.app";
+      host = "https://codegen.plasmic.app";
     } else {
-      return this.auth.host;
+      host = this.auth.host;
     }
+    console.log('__DEV__:codegenHost', { auth: this.auth, host, DEFAULT_HOST });
+    return host;
   }
 }
diff --git a/node_modules/@plasmicapp/cli/src/index.ts b/node_modules/@plasmicapp/cli/src/index.ts
index 1ee2787..f21d5bb 100644
--- a/node_modules/@plasmicapp/cli/src/index.ts
+++ b/node_modules/@plasmicapp/cli/src/index.ts
@@ -341,6 +341,11 @@ function configureSyncArgs(
       type: "array",
       default: [],
     })
+    .option("host", {
+      describe: "Plasmic host to use",
+      type: "string",
+      default: "https://studio.plasmic.app",
+    })
     .option("force", {
       type: "boolean",
       describe: "Force sync to bypass specified version ranges.",
diff --git a/node_modules/@plasmicapp/cli/src/utils/auth-utils.ts b/node_modules/@plasmicapp/cli/src/utils/auth-utils.ts
index a72060c..868b8d0 100644
--- a/node_modules/@plasmicapp/cli/src/utils/auth-utils.ts
+++ b/node_modules/@plasmicapp/cli/src/utils/auth-utils.ts
@@ -241,7 +241,9 @@ function failAuth() {
 export async function getOrStartAuth(
   opts: CommonArgs & { host: string; enableSkipAuth?: boolean }
 ) {
+  console.log('__DEV__:getOrStartAuth', { opts });
   let auth = await getCurrentAuth(opts.auth);
+  console.log('__DEV__:getOrStartAuth', { auth });
   if (!auth && opts.enableSkipAuth) {
     return;
   }
diff --git a/node_modules/@plasmicapp/cli/src/utils/get-context.ts b/node_modules/@plasmicapp/cli/src/utils/get-context.ts
index e1a36d8..ed6bd30 100644
--- a/node_modules/@plasmicapp/cli/src/utils/get-context.ts
+++ b/node_modules/@plasmicapp/cli/src/utils/get-context.ts
@@ -8,7 +8,7 @@ import { logger } from "../deps";
 import { CommonArgs } from "../index";
 import { runNecessaryMigrations } from "../migrations/migrations";
 import { HandledError } from "../utils/error";
-import { getCurrentAuth } from "./auth-utils";
+import { getCurrentAuth, getOrStartAuth } from "./auth-utils";
 import {
   DEFAULT_HOST,
   findConfigFile,
@@ -298,7 +298,7 @@ async function resolveMissingFilesInConfig(
 }
 
 export async function getContext(
-  args: CommonArgs,
+  args: CommonArgs & { host: string },
   {
     enableSkipAuth = false,
     skipMissingFiles = false,
@@ -311,7 +311,7 @@ export async function getContext(
 ): Promise<PlasmicContext> {
   if (!args.baseDir) args.baseDir = process.cwd();
   const auth = enableSkipAuth
-    ? await getCurrentOrDefaultAuth(args)
+    ? await getCurrentOrDefaultAuth({ ...args, enableSkipAuth })
     : await getOrInitAuth(args);
 
   /** Sentry */
@@ -379,14 +379,15 @@ export async function getContext(
 /**
  * Use empty user/token to signify no auth (only returning to provide a default host).
  */
-export async function getCurrentOrDefaultAuth(args: CommonArgs) {
-  const auth = await getCurrentAuth(args.auth);
+export async function getCurrentOrDefaultAuth(args: CommonArgs & { host: string; enableSkipAuth?: boolean }) {
+  // const auth = await getCurrentAuth(args.auth);
+  const auth = await getOrStartAuth(args);
   if (auth) {
     return auth;
   }
 
   return {
-    host: DEFAULT_HOST,
+    host: args.host || DEFAULT_HOST,
     user: "",
     token: "",
   };`;
}
exports.makePlasmicPatch_plasmicapp_cli = makePlasmicPatch_plasmicapp_cli;
